specifications/service-registry.yaml

```diff
@@ -0,0 +1,279 @@
+openapi: 3.0.0
+info:
+  description: 
+    API example for service registry. The API provides methods of distributing
+    and searching metadata related to mobility services. The API provides a way
+    for the service providers to publish their services and the the MaaS-operators
+    and end users to query available services and the user API standards in any 
+    area of interest. The services can be filtered by used API specifications 
+    and the certification proviced by trusted third parties. The API also provides
+    methods for distributing the registered metadata between multiple instances
+    of the service.<br/>
+    <br/>
+    Copyright 2017 FLOU Ltd.
+  version: "0.1.0-oas3"
+  title: Service registry
+  contact:
+    name: FLOU
+    url: 'https://www.flou.io/'
+    email: sami.makinen@flou.io
+  license:
+    name: Apache License 2.0
+    url: 'https://github.com/maas-alliance/apis/LICENSE'
+tags:
+  - name: catalogs
+    description: Handle service catalog registration
+  - name: queries
+    description: Query registered services
+paths:
+  /catalogs/announce:
+    post:
+      tags:
+        - catalogs
+      summary: Add a new service catalog to the registry
+      description: 
+        Announce new service provider to be registered. The services area 
+        registered by providing a link to a service catalog file that specifies
+        the services and APIs provided by the service provides. The Service 
+        Registry will then update its internal database by fetching and parsing
+        the service catalog file.
+      operationId: announceCatalog
+      requestBody:
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              type: object
+              properties:
+                uri:
+                  description: Location of the service catalog resource to register
+                  type: string
+                  format: catalogUri
+              required:
+                - uri
+      responses:
+        '200':
+          description: Successful operation
+        '403':
+          description: Catalog announcement forbidden
+        '405':
+          description: Invalid input
+  /catalogs/list:
+    get:
+      tags:
+        - catalogs
+      summary: List registered service catalogs
+      description: 
+        List service catalog URIs registered with this service registry. The 
+        query can be used by other Service Registries to mirror the content of
+        another registry.
+      operationId: listCatalogs
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/catalog-listing'
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/catalog-listing'
+        '403':
+          description: Catalog listing forbidden
+  /services/find:
```

Comment:
This has some conceptual overlap with [maas-api.yaml's regions/contain](https://github.com/maas-alliance/apis/blob/master/specifications/maas-api.yaml#L64) endpoint. Would be good to unify the two, or at least make the differences clear.

Advantages of `/services/find`:

- More detailed input options with `certs`, `types`, `sort`, etc.
- More generic description of available services

Advantages of `/regions/contain`:

- Can specify coordinate pairs as input, e.g., to determine what service covers inter-city routing 
- More detailed/specific results of what is covered, e.g., what modes and depth of integration

At the least, I recommend to add a `maas-api` case to the `types` enums.

Reply:
As discussed in the call just now, the harmonisation between the two can go into a separate task for later.